File: TATMediaSDKDemo/TATMediaSDKDemo/TATDiscoverViewController.m
//
//  TATDiscoverViewController.m
//  TATMediaSDKDemo
//
//  Created by wuleslie on 2019/12/19.
//  Copyright © 2019 wuleslie. All rights reserved.
//

#import "TATDiscoverViewController.h"
#import <TATMediaSDK/TATMediaSDK.h>
#import "TATUserManager.h"
#import "TATMediaManager.h"
@import WebKit;

@interface TATDiscoverViewController ()

@property (nonatomic, strong) WKWebView *webView;
@property (nonatomic, strong) UIView *bannerView;
@property (nonatomic, strong) UIView *floatView;

@end

@implementation TATDiscoverViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.view.backgroundColor = [UIColor colorWithRed:119.0/255 green:211.0/255 blue:220.0/255 alpha:1.0]; // 61 162 132
    self.navigationItem.title = @"发现";
    [self configWebView];
}

- (void)viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];
    [self showBannerAd];
    [self showFloatAd];
}

- (void)showBannerAd {
    [self.bannerView removeFromSuperview];
    __weak typeof(self)weakself = self;
    self.bannerView = [TATMediaCenter initSimpleAdWithSlotId:[TATMediaManager slotIdForType:TATSimpleAdTypeBanner] resultBlock:^(BOOL result, NSError *error) {
        if (result) {
            __strong typeof(weakself)self = weakself;
            CGRect frame = self.bannerView.frame;
            CGFloat originX = ([UIScreen mainScreen].bounds.size.width - frame.size.width) / 2;
            frame.origin.x = originX;
            frame.origin.y = 88 + 16;
            self.bannerView.frame = frame;
            [self relayoutWebView];
        } else {
            
        }
    }];
    [self.view addSubview:self.bannerView];
}

- (void)configWebView {
    self.webView = [[WKWebView alloc] initWithFrame:CGRectZero];
    // self.webView.navigationDelegate = self;
    NSURL *webURL = [NSURL URLWithString: @"https://book.douban.com"];
    [self.webView loadRequest:[NSURLRequest requestWithURL:webURL]];
    [self.view addSubview:self.webView];
}

- (void)relayoutWebView {
    CGFloat originY = CGRectGetMaxY(self.bannerView.frame) + 16;
    CGRect frame = CGRectMake(0, originY, [[UIScreen mainScreen] bounds].size.width, [[UIScreen mainScreen] bounds].size.height - originY);
    self.webView.frame = frame;
}

- (void)showFloatAd {
    [self.floatView removeFromSuperview];

    self.floatView = [TATMediaCenter initSimpleAdWithSlotId:[TATMediaManager slotIdForType:TATSimpleAdTypeFloat] resultBlock:^(BOOL result, NSError *error) {
        if (result) {
            CGRect frame = self.floatView.frame;
            CGFloat originX = ([UIScreen mainScreen].bounds.size.width - frame.size.width) - 16;
            frame.origin.x = originX;
            frame.origin.y = [UIScreen mainScreen].bounds.size.height - frame.size.height - 116;
            self.floatView.frame = frame;
        } else {
            
        }
    }];
    [self.view addSubview:self.floatView];
}


@end

File: TATSwiftDemo/TATSwiftDemo/TATDiscoverViewController.swift
//
//  TATDiscoverViewController.swift
//  TATSwiftDemo
//
//  Created by zhen yang on 2021/2/1.
//

import UIKit
import TATMediaSDK
import WebKit
class TATDiscoverViewController: UIViewController {
    var webView : WKWebView?
    var bannerView : UIView?
    var floatView : UIView?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.view.backgroundColor = UIColor.init(red: 119.0/255.0, green: 211.0/255.0, blue: 220.0/255.0, alpha: 1.0)
        navigationItem.title = "发现"
        configWebView()
    }
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        showBannerAd()
        showFloatAd()
        
    }
    
    func configWebView() -> Void {
        self.webView = WKWebView(frame: CGRect.zero)
        let webURL  = URL(string: "https://book.douban.com")
        let webRequest = URLRequest(url: webURL!)
        self.webView?.load(webRequest)
        self.view.addSubview(self.webView!)
    }
    
    func showBannerAd() -> Void {
        let adView = TATMediaCenter.initSimpleAd(withSlotId: TATMediaManager.slotIdForType(.banner)) { [weak self](result, error) in
            if result {
                if var frame = self?.bannerView?.frame {
                    let originX = (UIScreen.main.bounds.width - frame.width) * 0.5
                    frame.origin.x = originX
                    frame.origin.y = 88 + 16
                    self?.bannerView?.frame = frame
                    self?.relayoutWebView()
                }
                
            }
        }
        self.bannerView?.removeFromSuperview()
        self.bannerView = nil
        self.view.addSubview(adView!)
        self.bannerView = adView
    }
    
    func showFloatAd() -> Void {
        let adView = TATMediaCenter.initSimpleAd(withSlotId: TATMediaManager.slotIdForType(.banner)) { [weak self](result, error) in
            if result {
                if var frame = self?.floatView?.frame {
                    let originX = (UIScreen.main.bounds.width - frame.width) * 0.5
                    frame.origin.x = originX
                    frame.origin.y = 88 + 16
                    self?.floatView?.frame = frame
                    self?.relayoutWebView()
                }
                
            }
        }
        self.floatView?.removeFromSuperview()
        self.floatView = nil
        self.view.addSubview(adView!)
        self.floatView = adView
    }
    
    func relayoutWebView() -> Void {
        if let maxY = self.bannerView?.frame.maxY {
            let originY = maxY + 16
            let frame = CGRect(x: 0, y: originY, width: UIScreen.main.bounds.width, height: UIScreen.main.bounds.height - originY)
            self.webView?.frame = frame
        }
    }
}
